app/scripts/controllers/network/provider-api-tests/shared-tests.js

```diff
@@ -339,41 +769,47 @@ export function testsForRpcMethodAssumingNoBlockParam(method) {
       });
     });
 
-    it(`causes a request to fail with a custom error if the request to Infura returns a ${httpStatus} response 5 times in a row`, async () => {
-      await withMockedInfuraCommunications(async (comms) => {
+    it(`causes a request to fail with a custom error if the request to the RPC endpoint returns a ${httpStatus} response 5 times in a row`, async () => {
+      await withMockedCommunications({ providerType }, async (comms) => {
         const request = { method };
 
         // The first time a block-cacheable request is made, the latest block
         // number is retrieved through the block tracker first. It doesn't
         // matter what this is — it's just used as a cache key.
         comms.mockNextBlockTrackerRequest();
-        comms.mockInfuraRpcCall({
+        comms.mockRpcCall({
           request,
           response: {
             error: 'Some error',
             httpStatus,
           },
           times: 5,
         });
-        const promiseForResult = withInfuraClient(
+        comms.mockNextBlockTrackerRequest();
+        const promiseForResult = withNetworkClient(
+          { providerType },
           async ({ makeRpcCall, clock }) => {
             return await waitForPromiseToBeFulfilledAfterRunningAllTimers(
               makeRpcCall(request),
               clock,
             );
           },
         );
-
-        await expect(promiseForResult).rejects.toThrow(
-          /^InfuraProvider - cannot complete request\. All retries exhausted\..+Gateway timeout/su,
-        );
+        const err =
+          providerType === 'infura'
+            ? buildInfuraClientRetriesExhaustedErrorMessage('Gateway timeout')
+            : buildJsonRpcEngineEmptyResponseErrorMessage(method);
+        await expect(promiseForResult).rejects.toThrow(err);
       });
     });
   });
 
-  ['ETIMEDOUT', 'ECONNRESET', 'SyntaxError'].forEach((errorMessagePrefix) => {
-    it(`retries the request to Infura up to 5 times if an "${errorMessagePrefix}" error is thrown while making the request, returning the successful result if there is one on the 5th try`, async () => {
-      await withMockedInfuraCommunications(async (comms) => {
+  // Both the Infura and fetch middleware detect ETIMEDOUT errors and will
+  // automatically retry the request to the RPC endpoint in question, but both
```

Comment:
Nit: could we move the "success on 5th try" case outside of this `if` block, if the behavior is the same?

Reply:
I see this applies to a few other cases as well (the other `ETIMEDOUT` test, and the 503/504 handling upon 5 successive failures).

Reply:
Yes absolutely. I can make this change today.

Reply:
Fixed in [6898d7e](https://github.com/MetaMask/metamask-extension/pull/16337/commits/6898d7e823a0a3db2b9869a583ae8f6a2aa458e5).